src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.hpp

```diff
@@ -78,6 +81,17 @@ class ShenandoahHeuristics : public CHeapObj<mtGC> {
 protected:
   static const uint Moving_Average_Samples = 10; // Number of samples to store in moving averages
 
+  bool _start_gc_is_pending;              // True denotes that GC has been triggered, so no need to trigger again.
+  size_t _declined_trigger_count;         // This counts how many times since previous GC finished that this
+                                          //  heuristic has answered false to should_start_gc().
+  size_t _previous_trigger_declinations;  // This represents the value of _declined_trigger_count as captured at the
```

Comment:
Maybe the name should be _most_recent_declined_trigger_count, which relates it directly to _declined_trigger_count.

Reply:
Thanks for suggestion.  I'm making this change.